pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -177,16 +177,20 @@ private void refreshControllerLeaderMapFromLeadControllerResource() {
    * Updates lead controller pairs from Helix cluster leader.
    */
   private void refreshControllerLeaderMapFromHelixClusterLeader() {
-    Pair<String, Integer> helixClusterLeader = getHelixClusterLeader();
-    if (helixClusterLeader == null) {
-      _cachedControllerLeaderValid = false;
-      return;
-    }
-    for (int i = 0; i < Helix.NUMBER_OF_PARTITIONS_IN_LEAD_CONTROLLER_RESOURCE; i++) {
-      _cachedControllerLeaderMap.put(i, helixClusterLeader);
+    boolean refreshSucceeded = false;
+    try {
+      Pair<String, Integer> helixClusterLeader = getHelixClusterLeader();
+      if (helixClusterLeader == null) {
+        return;
+      }
+      for (int i = 0; i < Helix.NUMBER_OF_PARTITIONS_IN_LEAD_CONTROLLER_RESOURCE; i++) {
+        _cachedControllerLeaderMap.put(i, helixClusterLeader);
+      }
+      refreshSucceeded = true;
+      LOGGER.info("Refreshed controller leader map successfully.");
+    } finally {
+      _cachedControllerLeaderValid = refreshSucceeded;
```

Comment:
Please add a log if the refresh failed for any reason.

Reply:
Added.